scripts/macros/spells/spiritOfDeath.js

```diff
@@ -0,0 +1,221 @@
+        duration: 3600,
+        range: 90,
+        animation,
+        initiativeType: 'follows',
+        additionalSummonVaeButtons: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData].map(i => {return {type: 'use', name: i.name, identifier: i.flags['chris-premades'].info.identifier};})
+    });
+}
+async function attack({workflow}) {
+    if (workflow.targets.size !== 1 || workflow.advantage) return;
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'spiritOfDeathHauntCreature');
+    if (!effect) {
+        workflow.aborted = true;   
+        return;
+    }
+    let {targets: validTargetUuids, formula} = effect.flags['chris-premades'].spiritOfDeathHauntCreature;
+    if (!validTargetUuids.includes(workflow.targets.first().document.uuid)) {
+        workflow.aborted = true;
+        return;
+    }
+    workflow.advantage = true;
+    workflow.attackAdvAttribution.add(genericUtils.translate('DND5E.Advantage') + ': ' + effect.name);
+}
+async function late({workflow}) {
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'summonedEffect');
+    let originItem = fromUuidSync(effect?.origin);
+    let sourceActor = originItem?.actor;
+    if(!sourceActor)
+    {
+        return;
+    }
+    let featureData = await compendiumUtils.getItemFromCompendium(constants.packs.summonFeatures, 'Haunt Creature: Haunt', {object: true, getDescription: true, translate: 'CHRISPREMADES.Macros.SpiritOfDeath.HauntCreatureHaunt', identifier: 'spiritOfDeathHauntCreatureHaunt', flatDC: sourceActor.system.attributes.spelldc});
+    if (!featureData) {
+        errors.missingPackItem();
+        return;
+    }
+    let targetEffectData = {
+        name: genericUtils.translate('CHRISPREMADES.Macros.SpiritOfDeath.Haunted'),
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            dae: {
+                "specialDuration": [
+                    "zeroHP"
+                ]
+            }
+        }
+    };
+    let casterEffectData = {
+        name: workflow.item.name,
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            'chris-premades': {
+                spiritOfDeathHauntCreature: {
+                    targets: Array.from(workflow.targets).map(i => i.document.uuid),
+                }
+            }
+        }
+    };
+    let casterEffect = await effectUtils.createEffect(workflow.actor, casterEffectData, {concentrationItem: workflow.item, strictlyInterdependent: true, identifier: 'spiritOfDeathHauntCreature'});
+    await genericUtils.setFlag(casterEffect, 'chris-premades', 'macros.combat', ['spiritOfDeathHauntCreatureHaunt']);
+    for (let i of workflow.targets) {
+        if (i.actor) await effectUtils.createEffect(i.actor, targetEffectData, {strictlyInterdependent: true, parentEntity: casterEffect, identifier: 'spiritOfDeathHauntCreatureHaunted'});
```

Comment:
If _any_ of these creatures get un-haunted, should _all_ of them get un-haunted? If not, change `strictlyInterdependent` to just `interdependent`

Reply:
Textually, it's single target. But I started with just `interdependent` and it did not delete the parent effect on the spirit when I removed the child from the haunted creature, and when I added strictly it worked properly.

Reply:
Hm in retrospect it may be an interesting interaction where because concentration is dependent on the parent effect, "strictlyInterdependent" is the only one that works. Fine by me